### ISSUE TITLE
`uv add textract` fails with parse error on requires

### ISSUE THREAD

**Issue by DRMacIver** [astral-sh/uv#9194]
Running `uv add textract==1.6.4` on uv version `0.3.2` (on OSX if relevant) produces the following error:

```
⠹ textract==1.6.4                                                                                                                                                                                                     error: Failed to download and build `textract==1.6.4`
  Caused by: Build backend failed to determine extra requires with `build_wheel()` with exit status: 1
--- stdout:

--- stderr:
error in textract setup command: 'install_requires' must be a string or iterable of strings containing valid project/version requirement specifiers; .* suffix can only be used with `==` or `!=` operators
    extract-msg<=0.29.*
               ~~~~~~~^
---
```

The package in question pip installs fine.

This is presumably just the uv requirement parser being stricter than pip's, and this particular requirements construct is just sufficiently unusual to see in the wild that it hasn't come up before.

**Comment by DRMacIver** [comment]
If the error is purely setuptools related, why does pip install work with the same package specifier?

**Comment by charliermarsh** [comment]
Are pip and uv using the same version of Python? Is the package already cached by pip from some other install on your machine? Do you see the same behavior with `pip install --use-pep517 --no-cache --force-reinstall`?

**Comment by DRMacIver** [comment]
Same version of Python. Also correctly installs with `uv pip install`. Not cached. But indeed `--use-pep517` does cause pip to exhibit the same error.

**Comment by konstin** [comment]
For me locally, `pip install textract==1.6.4` errors on both Python 3.10 and Python 3.13 on ubuntu 24.04. My guess would be that the package is still in the pip cache, and was built before https://github.com/pypa/packaging/pull/407, which was released a `packaging` [22.0](https://github.com/pypa/packaging/releases/tag/22.0) which was merged in `setuptools` [67.0.0](https://setuptools.pypa.io/en/latest/history.html#v67-0-0 ). Before packaging 22.0 pip and setuptools were much more lenient with version specifiers, afterwards they started only accepting spec-compliant versions.

**Comment by DRMacIver** [comment]
Fair enough. In fact, I went to file a bug at textract and discovered this is already filed with them as an issue: https://github.com/deanmalmgren/textract/issues/533 (I didn't think to check there first because it seemed to work with pip).

Not that this necessarily matters to you, but it doesn't look very likely that's going to be fixed any time soon, as the package appears... lightly maintained. It would be good to have some way to work around this problem for similar issues, but that probably has to be on the setuptools side.
